js/lex.go

```diff
@@ -374,46 +377,57 @@ func (l *Lexer) consumeSingleLineComment() {
 
 ////////////////////////////////////////////////////////////////
 
-func (l *Lexer) consumeCommentToken() bool {
+func (l *Lexer) consumeHTMLLikeCommentToken() bool {
+	c := l.r.Peek(0)
+	if c == '<' && l.r.Peek(1) == '!' && l.r.Peek(2) == '-' && l.r.Peek(3) == '-' {
+		// opening HTML-style single line comment
+		l.r.Move(4)
+		l.consumeSingleLineComment()
+	} else if l.emptyLine && c == '-' && l.r.Peek(1) == '-' && l.r.Peek(2) == '>' {
+		// closing HTML-style single line comment
+		// (only if current line didn't contain any meaningful tokens)
+		l.r.Move(3)
+		l.consumeSingleLineComment()
+	} else {
+		return false
+	}
+	return true
+}
+
+func (l *Lexer) consumeRegularCommentToken() TokenType {
```

Comment:
I prefer to leave out `Regular`

Reply:
I thought about it, but figured it might be confusing that there is also another comment function that is not  called as part of this one with generic name. You think it won't be an issue?